agent/unit-test/config_unittest.cpp

```diff
@@ -60,505 +66,1038 @@ LOG_DECLARE_CLASS_CONTEXT("CFGL", "SoftwareContainer general config loader");
+                                     "sc.foo=1\n";
+
+    std::unique_ptr<ConfigLoader> loader(new StringConfigLoader(configString));
 
-    std::unique_ptr<ConfigLoaderAbstractInterface> loader(new StringConfigLoader(configString));
+    TypeMap typeMapping {{std::make_pair("SoftwareContainer", "sc.foo"),
+                          ConfigType::Integer}};
 
-    std::map<std::string, int> intOptions;
-    intOptions.insert(std::pair<std::string, int>("sc.preload", 2));
+    MainConfigSource source(std::move(loader), typeMapping);
 
-    Config config(std::move(loader),
-                  std::move(emptyDefaultConfig()),
-                  std::map<std::string, std::string>(),
-                  intOptions,
-                  std::map<std::string, bool>());
+    IntConfig expectedItem("SoftwareContainer", "sc.foo", 1);
 
-    ASSERT_TRUE(config.getIntegerValue("SoftwareContainer", "sc.preload") == 2);
+    std::vector<IntConfig> configs = source.intConfigs();
+
+    ASSERT_EQ(configs.back().key(), expectedItem.key());
+    ASSERT_EQ(configs.back().group(), expectedItem.group());
+    ASSERT_EQ(configs.back().value(), expectedItem.value());
 }
 
 /*
- * As above but for integer values
+ * Test that the expected string config items are returned by the source
+ *
+ * TODO: Test this with more configs
  */
-TEST_F(ConfigTest, ExplicitBoolValuesTakesPrecedence) {
+TEST_F(MainConfigSourceTest, ReturnsExpecedBoolConfigs) {
     // Well formed config
     const std::string configString = "[SoftwareContainer]\n"
-                                     "sc.shut-down-containers = true\n";
+                                     "sc.foo=false\n";
+
+    std::unique_ptr<ConfigLoader> loader(new StringConfigLoader(configString));
 
-    std::unique_ptr<ConfigLoaderAbstractInterface> loader(new StringConfigLoader(configString));
+    TypeMap typeMapping {{std::make_pair("SoftwareContainer", "sc.foo"),
+                          ConfigType::Boolean}};
 
-    std::map<std::string, bool> boolOptions;
-    boolOptions.insert(std::pair<std::string, bool>("sc.shut-down-containers", true));
+    MainConfigSource source(std::move(loader), typeMapping);
 
-    Config config(std::move(loader),
-                  std::move(emptyDefaultConfig()),
-                  std::map<std::string, std::string>(),
-                  std::map<std::string, int>(),
-                  boolOptions);
+    BoolConfig expectedItem("SoftwareContainer", "sc.foo", false);
 
-    ASSERT_TRUE(config.getBooleanValue("SoftwareContainer", "sc.shut-down-containers") == true);
+    std::vector<BoolConfig> configs = source.boolConfigs();
+
+    ASSERT_EQ(configs.back().key(), expectedItem.key());
+    ASSERT_EQ(configs.back().group(), expectedItem.group());
+    ASSERT_EQ(configs.back().value(), expectedItem.value());
 }
 
-/*
- * Test that Config throws an exception if a config dependency is not met
- *
- * TODO: Fix this when dependecies are implemented
- */
+
+
+
+
+
+
+
+
+
+// /*
```

Comment:
Is it possible to save this somewhere? I'm not sure where, but unless they are test cases which fail or even are expected to fail or need some refactoring to work... then it feels a bit strange for me to keep them. You mentioned something about them. Is it that you are using them for debugging while developing? Perhaps they can be moved to another test suite which can be removed when not needed anymore? Unless they will be removed very soon.